tesseract_environment/src/core/environment.cpp

```diff
@@ -161,227 +162,159 @@ bool Environment::isInitialized() const { return initialized_; }
-    {
-      CONSOLE_BRIDGE_logError("Unhandled environment command: CHANGE_LINK_ORIGIN");
-      assert(false);
-      return false;
-    }
-    case tesseract_environment::CommandType::CHANGE_JOINT_ORIGIN:
-    {
-      const auto& cmd = static_cast<const tesseract_environment::ChangeJointOriginCommand&>(command);
-      if (!changeJointOrigin(cmd.getJointName(), cmd.getOrigin()))
-        return false;
-      break;
-    }
-    case tesseract_environment::CommandType::CHANGE_LINK_COLLISION_ENABLED:
-    {
-      const auto& cmd = static_cast<const tesseract_environment::ChangeLinkCollisionEnabledCommand&>(command);
-      setLinkCollisionEnabled(cmd.getLinkName(), cmd.getEnabled());
-      if (getLinkCollisionEnabled(cmd.getLinkName()) != cmd.getEnabled())
-        return false;
-      break;
-    }
-    case tesseract_environment::CommandType::CHANGE_LINK_VISIBILITY:
-    {
-      const auto& cmd = static_cast<const tesseract_environment::ChangeLinkVisibilityCommand&>(command);
-      setLinkVisibility(cmd.getLinkName(), cmd.getEnabled());
-      if (getLinkVisibility(cmd.getLinkName()) != cmd.getEnabled())
-        return false;
-      break;
-    }
-    case tesseract_environment::CommandType::ADD_ALLOWED_COLLISION:
-    {
-      const auto& cmd = static_cast<const tesseract_environment::AddAllowedCollisionCommand&>(command);
-      addAllowedCollision(cmd.getLinkName1(), cmd.getLinkName2(), cmd.getReason());
-      break;
-    }
-    case tesseract_environment::CommandType::REMOVE_ALLOWED_COLLISION:
-    {
-      const auto& cmd = static_cast<const tesseract_environment::RemoveAllowedCollisionCommand&>(command);
-      removeAllowedCollision(cmd.getLinkName1(), cmd.getLinkName2());
-      break;
-    }
-    case tesseract_environment::CommandType::REMOVE_ALLOWED_COLLISION_LINK:
-    {
-      const auto& cmd = static_cast<const tesseract_environment::RemoveAllowedCollisionLinkCommand&>(command);
-      removeAllowedCollision(cmd.getLinkName());
-      break;
-    }
-    case tesseract_environment::CommandType::ADD_SCENE_GRAPH:
-    {
-      const auto& cmd = static_cast<const tesseract_environment::AddSceneGraphCommand&>(command);
-      if (cmd.getJoint() != nullptr)
+      case tesseract_environment::CommandType::MOVE_JOINT:
       {
-        if (!addSceneGraph(*(cmd.getSceneGraph()), cmd.getJoint()->clone(), cmd.getPrefix()))
-          return false;
+        auto cmd = std::dynamic_pointer_cast<const MoveJointCommand>(command);
+        success &= applyMoveJointCommand(cmd);
+        break;
       }
-      else
+      case tesseract_environment::CommandType::REMOVE_LINK:
       {
-        // This should only occur if the this graph is empty.
-        if (!addSceneGraph(*(cmd.getSceneGraph()), cmd.getPrefix()))
-          return false;
+        auto cmd = std::dynamic_pointer_cast<const RemoveLinkCommand>(command);
+        success &= applyRemoveLinkCommand(cmd);
+        break;
```

Comment:
Would it be unsafe to make these static_pointer_casts? The only way it could not be that type is if we have a bug in the code.

Reply:
I think this would be ok.

Reply:
done.